src/main/controllers/features/limited/companyLookupController.ts

```diff
@@ -6,18 +6,35 @@ import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../.
 import { BASE_URL, LIMITED_IS_THIS_YOUR_COMPANY, LIMITED_WHAT_IS_THE_COMPANY_NUMBER, TYPE_OF_BUSINESS } from "../../../types/pageURL";
 import { CompanyLookupService } from "../../../services/companyLookupService";
 import { formatValidationError, getPageProperties } from "../../../validation/validation";
+import { GET_ACSP_REGISTRATION_DETAILS_ERROR, SUBMISSION_ID } from "../../../common/__utils/constants";
+import logger from "../../../../../lib/Logger";
+import { getAcspRegistration } from "../../../services/acspRegistrationService";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
 
     const lang = selectLang(req.query.lang);
     const locales = getLocalesService();
+    const session: Session = req.session as any as Session;
+    try {
+        const acsp = await getAcspRegistration(session, session.getExtraData(SUBMISSION_ID)!, res.locals.userEmail);
+        session.setExtraData("typeOfBusinessService", acsp.typeOfBusiness);
```

Comment:
We should save the data in ACSP data model session variable. We will not need to then create another session variable typeOfBusinessService

Reply:
I have added this just to demonstrate the capability. There is a separate ticket for integration part where we can cater this.